owsi-core/owsi-core-components/owsi-core-component-wicket-more/src/main/java/fr/openwide/core/wicket/more/markup/html/repeater/data/table/builder/action/factory/AbstractActionColumnElementFactory.java

```diff
@@ -0,0 +1,165 @@
+package fr.openwide.core.wicket.more.markup.html.repeater.data.table.builder.action.factory;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+import org.apache.wicket.markup.html.link.AbstractLink;
+import org.apache.wicket.model.IModel;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.impossibl.postgres.utils.guava.Joiner;
+
+import fr.openwide.core.wicket.more.condition.Condition;
+import fr.openwide.core.wicket.more.markup.html.bootstrap.label.renderer.BootstrapRenderer;
+import fr.openwide.core.wicket.more.markup.html.factory.IOneParameterComponentFactory;
+import fr.openwide.core.wicket.more.markup.html.factory.IOneParameterConditionFactory;
+import fr.openwide.core.wicket.more.util.model.Detachables;
+
+
+public abstract class AbstractActionColumnElementFactory<T, F extends AbstractActionColumnElementFactory<T, ?>> implements IOneParameterComponentFactory<AbstractLink, IModel<T>> {
+
+	private static final long serialVersionUID = 8791565179874571105L;
+
+	private final BootstrapRenderer<? super T> renderer;
+
+	private Condition showLabelCondition = Condition.alwaysTrue();
```

Comment:
`import com.impossibl.postgres.utils.guava.Joiner`

-> once more, be careful with the imports.